opentracing-zipkin-v2/OpenTracing/Zipkin/V2/HttpReporter.hs

```diff
@@ -144,8 +146,15 @@ spanE loc logfmt s = pairs $
              (pair "remoteEndpoint")
              (view (spanTags . to remoteEndpoint) s)
     <> pair "annotations"    (list (logRecE logfmt) $ view spanLogs s)
-    <> pair "tags"           (toEncoding $ view spanTags s)
+    -- Zipkin V2 requires tag values to be strings
+    <> pair "tags"           (toEncoding . toMapOf (spanTags . to fromTags . ifolded . to tagToText) $ s)
     -- nb. references are lost, perhaps we should stick them into annotations?
+  where tagToText = \ case
```

Comment:
I'd prefer to use `toEncoding` here (defined in `OpenTracing.Tags`), because that's potentially more efficient than going through `pack . show`, and also it properly base64-encodes binary values.

Reply:
Amending this comment: it seems better to forgo type classes entirely. The `show` still bugs me, though...

Reply:
Sorry, not using the typeclass is a good idea. Also, looking closer, `show` is probably fine too (look at `Data.ByteString.Builder.doubleDec`, which aeson uses :0). The only thing would be to base64-encode the `BinaryT` case. I'll just merge and make that change myself, hope you don't mind.

Reply:
Awesome, thank you!!